records_mover/records/validated_records_hints.py

```diff
@@ -0,0 +1,151 @@
+from typing import NamedTuple, Union, Literal, Optional, TypeVar, List
+from .types import (RecordsHints, HintHeaderRow, HintFieldDelimiter,
+                    HintCompression, HintRecordTerminator,
+                    HintQuoting, HintQuoteChar, HintDoublequote,
+                    HintEscape, HintEncoding, HintDateFormat,
+                    HintTimeOnlyFormat, HintDateTimeFormatTz,
+                    HintDateTimeFormat)
+from .types import (VALID_COMPRESSIONS, VALID_QUOTING, VALID_ESCAPE,
+                    VALID_ENCODING, VALID_DATEFORMATS, VALID_TIMEONLYFORMATS,
+                    VALID_DATETIMEFORMATTZS, VALID_DATETIMEFORMATS)
+from .hints import cant_handle_hint
+
+
+# TODO: Read up on namedtuple vs dataclass
+# TODO: Read up on autovalidating libraries
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker